agent/plugin.go

```diff
@@ -86,18 +86,30 @@ func CreatePluginsFromJSON(j string) ([]*Plugin, error) {
 			plugins = append(plugins, plugin)
 		case map[string]interface{}:
 			for location, config := range vv {
-				// Ensure the config is a hash
-				config, ok := config.(map[string]interface{})
-				if !ok {
-					return nil, fmt.Errorf("Configuration for \"%s\" is not a hash", location)
+				// First check to make sure this plugin has config
```

Comment:
Took me a while to parse this, might it be clearer with a `if config == nil` block up the top for handling the special case then a continue rather than `if/else`?